membership-attribute-service/app/controllers/ExistingPaymentOptionsController.scala

```diff
@@ -43,12 +44,12 @@ class ExistingPaymentOptionsController(
       maybeUserId: Option[String],
       contactRepository: ContactRepository,
       subscriptionService: SubscriptionService,
-  ): SimpleEitherT[Map[AccountId, List[Subscription[SubscriptionPlan.AnyPlan]]]] =
+  ): SimpleEitherT[Map[Account, List[(Account, Subscription[SubscriptionPlan.AnyPlan])]]] =
     (for {
       user <- ListTEither.fromOption(maybeUserId)
       contact <- ListTEither.fromFutureOption(contactRepository.get(user))
       subscription <- ListTEither.fromFutureList(subscriptionService.since[SubscriptionPlan.AnyPlan](date)(contact))
-    } yield subscription).toList.map(_.groupBy(_.accountId))
+    } yield subscription).toList.map(_.groupBy { case (account, sub) => account })
```

Comment:
I changed this groupBy to use the account rather than the ID, so that I could access the account later in place of the account ID. However, assuming the account comparison is a full equality test, this is inefficient: I’d rather group by their account ID but keep the first encountered account in each group (effectively assuming uniqueness of zuora’s account IDs, which seems fair). I suppose if the grouping could produce a non-empty list type this would be straightforward.